Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h

```diff
@@ -53,15 +53,17 @@ class DetGeomDesc {
   using RotationMatrix = ROOT::Math::Rotation3D;
   using Translation = ROOT::Math::DisplacementVector3D<ROOT::Math::Cartesian3D<double>>;
 
+  DetGeomDesc() = default;
+
   // Constructor from old DD DDFilteredView
   DetGeomDesc(const DDFilteredView& fv);
   // Constructor from DD4Hep DDFilteredView
   DetGeomDesc(const cms::DDFilteredView& fv);
 
-  DetGeomDesc(const DetGeomDesc&);
-  DetGeomDesc& operator=(const DetGeomDesc&);
   virtual ~DetGeomDesc();
 
+  DetGeomDesc* makeCopyWithoutChildren() const;
```

Comment:
Can this be made to return `std::unique_ptr<DetGeomDesc>` to make it explicit that the caller has the ownership?

Reply:
Looks like a return by value would work as well. If so, that is better than returning by pointer or unique_ptr.

Reply:
The current code works with pointers:

https://github.com/cms-sw/cmssw/blob/0e26c4a0f7658a37176349a8e446492503cdf81c/Geometry/VeryForwardGeometryBuilder/interface/DetGeomDesc.h#L52

https://github.com/cms-sw/cmssw/blob/0e26c4a0f7658a37176349a8e446492503cdf81c/Geometry/VeryForwardGeometryBuilder/plugins/CTPPSGeometryESModule.cc#L131

thus I proposed a function returning a pointer. Switching to unique_ptr is trivial. Let me know what your suggestion is.

Reply:
One option would be to add a (private) constructor that takes all
https://github.com/cms-sw/cmssw/blob/0e26c4a0f7658a37176349a8e446492503cdf81c/Geometry/VeryForwardGeometryBuilder/src/DetGeomDesc.cc#L65-L76
as parameters.

Reply:
I was thinking it might be better to remove `makCopyWithoutChildren` and create a constructor that takes a `DetGeomDesc const&` and a 'dummy' enum with the only value `noChildren`. Then the calling code would become
```C++
DetGeomDesc* alignedDetRoot = new DetGeomDesc(idealDetRoot, DetGeomDesc::noChildren);
```

Reply:
@Dr15Jones Yes exactly
+10000

Reply:
Many thanks for all the suggestions! I also like this one. I will give it a try tomorrow.